src/dotnet/commands/dotnet-publish/PublishCommand.cs

```diff
@@ -416,7 +416,10 @@ private IEnumerable<ProjectContext> SelectContexts(string projectPath, NuGetFram
                 throw new InvalidProjectException($"'{projectPath}' does not contain a project.json file");
             }
 
-            var contexts = Workspace.GetProjectContextCollection(projectPath).FrameworkOnlyContexts;
+            var contextCollection = Workspace.GetProjectContextCollection(projectPath);
+            contextCollection.EnsureValid(projectPath);
```

Comment:
If you make EnsureValid return the ProjectContextCollection, add .FrameworkOnlyContexts here.